app/src/main/java/com/android/developers/androidify/navigation/MainNavigation.kt

```diff
@@ -110,6 +113,51 @@ fun MainNavigation() {
                     onAboutPressed = {
                         backStack.add(About)
                     },
+                    onImageCreated = { resultImageUri, prompt, originalImageUri ->
+                        backStack.removeAll{ it is ImageResult}
+                        backStack.add(
+                            ImageResult(
+                                result = resultImageUri.toString(),
+                                prompt = prompt,
+                                originalImageUri = originalImageUri?.toString()
+                            )
+                        )
+                    }
+                )
+            }
+            entry<ImageResult> { resultKey ->
+                ResultsScreen(
+                    resultImageUri = resultKey.result.toUri(),
+                    originalImageUri = resultKey.originalImageUri?.toUri(),
+                    onNextPress = { resultImageUri, originalImageUri ->
+                        backStack.removeAll{ it is ImageResult}
+                        backStack.add(
+                            ShareResult(
+                                resultUri = resultImageUri.toString(),
+                                originalImageUri = originalImageUri?.toString()
+                            )
+                        )
+                    },
+                    promptText = resultKey.prompt,
+                    onAboutPress = {
+                        backStack.add(About)
+                    },
+                    onBackPress = {
+                        backStack.removeLastOrNull()
+                        backStack.add(Create(fileName = resultKey.originalImageUri, prompt = resultKey.prompt))
```

Comment:
As far as I understand, we shouldn't need to do this as the Create() entry should already be on the backstack after removing the last one?